src/main/java/frc/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -0,0 +1,264 @@
+        }
+        return new SwerveModuleState(
+                targetSpeed, Rotation2d.fromDegrees(targetAngle));
+    }
+
+    /**
+     * @param scopeReference Current Angle
+     * @param newAngle       Target Angle
+     * @return Closest angle within scope
+     */
+    private static double placeInAppropriate0To360Scope(
+            double scopeReference, double newAngle) {
+        double lowerBound;
+        double upperBound;
+        double lowerOffset = scopeReference % 360;
+        if(lowerOffset >= 0) {
+            lowerBound = scopeReference - lowerOffset;
+            upperBound = scopeReference + (360 - lowerOffset);
+        } else {
+            upperBound = scopeReference - lowerOffset;
+            lowerBound = scopeReference - (360 + lowerOffset);
+        }
+        while(newAngle < lowerBound) {
+            newAngle += 360;
+        }
+        while(newAngle > upperBound) {
+            newAngle -= 360;
+        }
+        if(newAngle - scopeReference > 180) {
+            newAngle -= 360;
+        } else if(newAngle - scopeReference < -180) {
+            newAngle += 360;
+        }
+        return newAngle;
+    }
+
+    public void setDesiredState(SwerveModuleState desiredState, boolean isOpenLoop) {
+        // Custom optimize command, since default WPILib optimize assumes
+        // continuous controller which CTRE is not
+        desiredState = optimize(
+                desiredState,
+                getState().angle
+        );
+        if(isOpenLoop) {
+            /* If we're in open loop, we don't want to use the PID controller,
+            and we just set the drive speed in percentage of the max speed */
+            double percentOutput =
+                    desiredState.speedMetersPerSecond / SwerveConstants.MAX_SPEED;
+            mDriveMotor.set(
+                    ControlMode.PercentOutput,
+                    percentOutput / SwerveConstants.SPEED_DIVIDER);
+        } else {
+            /* If we're in closed loop, we want to use the PID controller,
+            and we set the drive speed in meters per second  */
+            double velocity = EncoderConversions.MPSToFalcon(
+                    desiredState.speedMetersPerSecond,
+                    SwerveConstants.WHEEL_CIRCUMFERENCE,
+                    SwerveConstants.DRIVE_GEAR_RATIO);
+            // Sets the drive motor velocity with the driveFeedforward.
+            mDriveMotor.set(
+                    ControlMode.Velocity, velocity, DemandType.ArbitraryFeedForward,
+                    driveFeedforward.calculate(desiredState.speedMetersPerSecond));
+        }
+
+        /*//Prevent rotating module if speed is less than 1%. Prevents jitter.
+        double desiredAngle =
+                (Math.abs(desiredState.speedMetersPerSecond) <= (SwerveConstants
+                .MAX_SPEED * 0.01)) ?
+                lastAngle :
+                desiredState.angle.getDegrees(); */
```

Comment:
Why do we not use this? last year it was an issue

Reply:
I don't see the point for this. I don't recall it being an issue last year, it's not an issue right now anyway

Reply:
it was an issue last year. remember how the modules returned to 0 when you didn't move. the way they wrote their code must have fixed it.